pkg/ext/stores/tokens/tokens.go

```diff
@@ -0,0 +1,844 @@
+package tokens
+
+import (
+	ext "github.com/rancher/rancher/pkg/apis/ext.cattle.io/v1"
+	extcore "github.com/rancher/steve/pkg/ext"
+
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/rancher/rancher/pkg/auth/tokens/hashers"
+	v3 "github.com/rancher/rancher/pkg/generated/controllers/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/wrangler"
+	v1 "github.com/rancher/wrangler/v3/pkg/generated/controllers/core/v1"
+	"github.com/rancher/wrangler/v3/pkg/randomtoken"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+	"k8s.io/apiserver/pkg/registry/rest"
+)
+
+const (
+	TokenNamespace = "cattle-tokens"
+	ThirtyDays     = 30 * 24 * 60 * 60 * 1000 // 30 days in milliseconds.
+	UserIDLabel    = "authn.management.cattle.io/token-userId"
+
+	// data fields used by the backing secrets to store token information
+	fieldAuthProvider   = "auth-provider"
+	fieldClusterName    = "cluster-name"
+	fieldDescription    = "description"
+	fieldDisplayName    = "display-name"
+	fieldEnabled        = "enabled"
+	fieldHash           = "hash"
+	fieldIsLogin        = "is-login"
+	fieldLastUpdateTime = "last-update-time"
+	fieldLastUsedAt     = "last-used-at"
+	fieldLoginName      = "login-name"
+	fieldPrincipalID    = "principal-id"
+	fieldTTL            = "ttl"
+	fieldUID            = "kube-uid"
+	fieldUserID         = "user-id"
+	// fieldIdleTimeout = "idle-timeout"	FUTURE ((USER ACTIVITY))
+
+	noUpdatePermission      = 0
+	fullUpdatePermission    = 1
+	limitedUpdatePermission = 2
+)
+
+// +k8s:openapi-gen=false
+// +k8s:deepcopy-gen=false
+
+// TokenStore is the interface to the token store seen by the extension API and users. Wrapped
+// around a SystemTokenStore it performs the necessary checks to ensure that Users have only access
+// to the tokens they are permitted to.
+type TokenStore struct {
+	SystemTokenStore
+}
+
+// +k8s:openapi-gen=false
+// +k8s:deepcopy-gen=false
+
+// SystemTokenStore is the interface to the token store used internally by other parts of
+// rancher. It does not perform any kind of permission checks, and operates with (implied) admin
+// authority. IOW it has access to all the tokens, in all ways.
+type SystemTokenStore struct {
+	namespaceClient     v1.NamespaceClient   // access to namespaces
+	support             supportActionHandler // subsystem for permission checking, secret value and hashing, timing
+	secretClient        v1.SecretClient
+	userAttributeClient v3.UserAttributeClient
+	userClient          v3.UserClient
+	initialized         bool
+}
+
+func NewTokenStoreFromWrangler(wranglerContext *wrangler.Context) extcore.Store[*ext.Token, *ext.TokenList] {
```

Comment:
Same here. It can be just `New()` or `NewSystem()` for system store. It's already in `stores/tokens` package.

Reply:
Done.